bench_test.go

```diff
@@ -0,0 +1,264 @@
+package yarpc_test
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/yarpc/yarpc-go"
+	"github.com/yarpc/yarpc-go/encoding/raw"
+	"github.com/yarpc/yarpc-go/transport"
+	yhttp "github.com/yarpc/yarpc-go/transport/http"
+	ytchannel "github.com/yarpc/yarpc-go/transport/tchannel"
+
+	"github.com/stretchr/testify/require"
+	"github.com/uber/tchannel-go"
+	traw "github.com/uber/tchannel-go/raw"
+	"golang.org/x/net/context"
+	"golang.org/x/net/context/ctxhttp"
+)
+
+func yarpcEcho(reqMeta yarpc.ReqMeta, body []byte) ([]byte, yarpc.ResMeta, error) {
+	return body, yarpc.NewResMeta(reqMeta.Context()).Headers(reqMeta.Headers()), nil
+}
+
+var httpEcho http.HandlerFunc = func(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+	hs := w.Header()
+	for k, vs := range r.Header {
+		for _, v := range vs {
+			hs.Set(k, v)
+		}
+	}
+
+	_, err := io.Copy(w, r.Body)
+	if err != nil {
+		panic(fmt.Sprintf("HTTP handler failed: %v", err))
+	}
+}
+
+type tchannelEcho struct{ b *testing.B }
+
+func (tchannelEcho) Handle(ctx context.Context, args *traw.Args) (*traw.Res, error) {
+	return &traw.Res{Arg2: args.Arg2, Arg3: args.Arg3}, nil
+}
+
+func (t tchannelEcho) OnError(ctx context.Context, err error) {
+	t.b.Fatalf("request failed: %v", err)
+}
+
+func withDispatcher(b *testing.B, cfg yarpc.Config, f func(yarpc.Dispatcher)) {
+	d := yarpc.NewDispatcher(cfg)
+	require.NoError(b, d.Start(), "failed to start server")
+	defer d.Stop()
+
+	f(d)
+}
+
+func runYARPCClient(b *testing.B, c raw.Client) {
+	reqBody := []byte("hello")
```

Comment:
do you want to make this a global variable so we know all the tests are using the same body, `var _reqBody = [...]`